.github/workflows/other-tests.yml

```diff
@@ -389,6 +389,15 @@ jobs:
               cd e2e/bug-9574
               cd test
               ../../../phpstan
+          - php-version: 8.2
+            ini-values: memory_limit=256M
+            operating-system: ubuntu-latest
+            script: |
+              cd e2e/bug8778
+              composer dump
+              ../../phpstan clear-result-cache -vvv -q
+              ../../phpstan analyse -vvv -q
+              ! ../../phpstan analyse -vvv --generate-baseline 2>&1 | grep "Result cache not used because the metadata do not match"
```

Comment:
we grep for the debug output and because of the leading `!` of this line is the exit code of the last pipe inverted.

means we are getting only a successfull exit code, when the text was not matched

Reply:
This `grep` business can break really easily.

What I'd like instead:

1) Add `isResultCacheUsed` to AnalysisResult (pass `$resultCache->isFullAnalysis()` to the constructor)
2) Create `FailWithoutResultCacheErrorFormatter`. After it checks that result cache was used, we can just call `TableErrorFormatter`.
3) Use this error formatter here in this test instead of `grep`.